### PR TITLE
Evancarr/frontend user flow

### PR DESCRIPTION
**Pull request by ejrcarr**
### Description:

This PR includes the following changes:

* Designed and implemented the styles designed in Figma
* Added a placeholder JSON file to mock the information on our backend so that the POST and GET requests can work properly
* Mapped each application to its own page accessible by app/[id]

With these changes, users can now fill out the application form and submit it, with the information being added to our data. The GET request can be used to retrieve all applications or a single application based on ID. Additionally, the applications are now accessible through their own pages, which will enhance the user experience.

### Screenshots
Screen Shot 2023-04-26 at 12 48 37 PM
Screen Shot 2023-04-26 at 12 48 46 PM
Screen Shot 2023-04-26 at 12 48 58 PM
Screen Shot 2023-04-26 at 12 49 08 PM
Screen Shot 2023-04-26 at 12 49 16 PM
Screen Shot 2023-04-26 at 12 49 28 PM
Screen Shot 2023-04-26 at 12 49 39 PM

Please review and let me know if any further changes are needed.

### CODE REVIEW

File: frontend/src/app/layout.tsx [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
+import styles from '@/styles/pages/layout.module.css';
+import '../styles/global/globals.css';
 import { Metadata } from 'next';
+import { Raleway } from 'next/font/google';
 
-const inter = Inter({ subsets: ['latin'] });
+const raleway = Raleway({ subsets: ['latin'] });
 
 export const metadata: Metadata = {
-	title: 'Elon ODS',
```
Comment (jumarmartin) [inline frontend/src/app/layout.tsx:11]:
```suggestion
	title: 'Open Data Service',
```

File: frontend/src/app/layout.tsx [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
+	title: {
+		default: 'Elon ODS',
+		template: '%s | Elon ODS',
```
Comment (jumarmartin) [inline frontend/src/app/layout.tsx:13]:
```suggestion
		default: 'Open Data Service',
		template: '%s | ODS',
```

File: frontend/src/app/page.tsx [hunk 3]
```diff
@@ -25,6 +40,73 @@ export default function Home() {
...
 				<label htmlFor='validUser'>Valid User</label>
 			</div>
+			<section className={styles.accessibleDataSection}>
+				<div className={styles.dataSectionInfo}>
+					<h3 className={`${styles.subheading} ${styles.dataFormatSubheading}`}>
+						Data Formats
+					</h3>
+					<h2 className={styles.dataFormatHeading}>Easily Accessible Data</h2>
+					<p className={styles.dataFormatDescription}>
+						Elon ODS provides a simple and easy way to access comprehensive data
```
Comment (jumarmartin) [inline frontend/src/app/page.tsx:50]:
```suggestion
						Open Data Service provides a simple and easy way to access comprehensive data

File: frontend/src/ui/Footer/Footer.tsx [hunk 4]
```diff
@@ -3,8 +3,8 @@ import styles from './Footer.module.css';
 const Footer: React.FunctionComponent = () => {
 	return (
 		<footer className={styles.container}>
-			Built with 🖤 by the Elon Society of Computing Project Team in Partnership
-			with the Elon University Office of IT
+			{/* Built with 🖤 by the Elon Society of Computing Project Team in Partnership
```
Comment (jumarmartin) [inline frontend/src/ui/Footer/Footer.tsx:6]:
```suggestion
			{/* Built with ❤️ by the Elon Society of Computing Project Team in Partnership
```

File: frontend/src/ui/Navbar/Navbar.tsx [hunk 5]
```diff
@@ -1,10 +1,31 @@
...
+import { Raleway } from 'next/font/google';
+
+const raleway = Raleway({ subsets: ['latin'] });
 
 const Navbar: React.FunctionComponent = () => {
 	return (
 		<nav className={styles.container}>
-			<Link href='/'>Open Data @ Elon</Link>
+			<Link href='/' className={`${raleway.className} ${styles.logo}`}>
+				Elon<span className={styles.odsLogo}>ODS</span>
```
Comment (jumarmartin) [inline frontend/src/ui/Navbar/Navbar.tsx:11]:
```suggestion
				<span className={styles.odsLogo}>Open Data Service</span>
```

**Review by jumarmartin — COMMENTED** [review]
Looks good, made some suggestions; feel free to respond.

File: frontend/src/app/apps/[id]/page.tsx
[the remark this answers is not in the document]
Comment (ejrcarr) [reply frontend/src/app/apps/[id]/page.tsx:1]: If that is the only reason to life the calls up, can we just refer to the process.env.NODE_ENV to determine if we are in development or prod and then export the api depending on that? I have committed an example that seems to work, take a look and tell me if that was what you were referring to

File: frontend/src/ui/Footer/Footer.tsx (hunk 4 above)
Comment (ejrcarr) [reply frontend/src/ui/Footer/Footer.tsx:6]: Do we have to reserve the footer for the a display of partnership with the Elon University Office of IT? If not, a footer may not be necessary and we can add in in the docs or put it in the hero

File: frontend/src/app/api/applications/route.tsx
[the remark this answers is not in the document]
Comment (ejrcarr) [reply frontend/src/app/api/applications/route.tsx:1]: Not entirely certain I know what you mean, can you elaborate a bit further? Or maybe show an example?

File: frontend/src/app/api/applications/data.json
[the remark this answers is not in the document]
Comment (ejrcarr) [reply frontend/src/app/api/applications/data.json:1]: Was planning to include tests, good to know I can start now. I might have to consult with you on how to set up a file structure that includes tests

File: frontend/src/ui/Footer/Footer.tsx (hunk 4 above)
Comment (jumarmartin) [reply frontend/src/ui/Footer/Footer.tsx:6]: This isn't a negotiated requirement but to put forth that we're building in partnership with the institution does lead to our credbility.

File: frontend/src/app/apps/[id]/page.tsx
[the remark this answers is not in the document]
Comment (jumarmartin) [reply frontend/src/app/apps/[id]/page.tsx:1]: That's correct and would be encapsulated by the isProdOrLocal function. Actually, a better name might be `url` or `siteURL` which might be a keyword exposed by next. I'm not exactly sure on that.

File: frontend/src/app/api/applications/route.tsx
[the remark this answers is not in the document]
Comment (jumarmartin) [reply frontend/src/app/api/applications/route.tsx:1]: We are currently using `applications/data.json` as [our way of returning data](https://github.com/elonsoc/ods/pull/34/files/09469485808ed17ad599d06e3cb4e722af70d4e6#diff-1a611087fc9266a23237e1dd5c525893bf10653975e94e1a8461653ba0719686R7). We shouldn't refer to a [hard-coded list](https://github.com/elonsoc/ods/pull/34/files/09469485808ed17ad599d06e3cb4e722af70d4e6#diff-1a611087fc9266a23237e1dd5c525893bf10653975e94e1a8461653ba0719686R3) but instead create an array and returning that.

[the remark this answers is not in the document]
Comment (jumarmartin) [reply frontend/src/app/api/applications/route.tsx:1]: We are currently using `applications/data.json` as [our way of returning data](https://github.com/elonsoc/ods/pull/34/files/09469485808ed17ad599d06e3cb4e722af70d4e6#diff-1a611087fc9266a23237e1dd5c525893bf10653975e94e1a8461653ba0719686R7). We shouldn't refer to a [hard-coded list](https://github.com/elonsoc/ods/pull/34/files/09469485808ed17ad599d06e3cb4e722af70d4e6#diff-1a611087fc9266a23237e1dd5c525893bf10653975e94e1a8461653ba0719686R3) but instead create an array and returning that.

File: frontend/src/app/api/applications/data.json
[the remark this answers is not in the document]
Comment (jumarmartin) [reply frontend/src/app/api/applications/data.json:1]: Feel free to ping me when you add them :)

File: frontend/src/config/Constants.tsx
Comment (jumarmartin) [inline frontend/src/config/Constants.tsx:1]: I really like this interface for configuring stuff.

**Review by jumarmartin — APPROVED** [review]
LGTM, great work :) 👍

**Pull request merged**